src/teacher.rb

```diff
@@ -0,0 +1,12 @@
+require './person'
+# Teacher class demonstrating the inheritance principle
+class Teacher < Person
+  def initialize(specialization, name = 'Unknown', age = 0, parent_permission: true)
+    super(name, age, parent_permission)
```

Comment:
- [ ] Please make sure to pass the parameters in the right order to the `super` constructor. The `Person` constructor accepts the parameters in the following order:

```ruby
def initialize(age, name = 'Unknown', parent_permission: true)
```

- On the other hand, you're passing the parameters to it from the `super` constructor: this way:

```ruby
super(name, age, parent_permission)
```

Reply:
Hello @ShadyShawkat 

Thank for reviewing my code. A quick question how do i go around this if rubocop always flags an error for using the right order.